OrbitCore/SymbolHelper.h

```diff
@@ -5,27 +5,34 @@
 #ifndef SYMBOL_HELPER_H_
 #define SYMBOL_HELPER_H_
 
+#include <filesystem>
 #include <string>
 #include <vector>
 
 #include "OrbitBase/Result.h"
 #include "symbol.pb.h"
 
 class SymbolHelper {
+  using path = std::filesystem::path;
+
  public:
   SymbolHelper();
-  explicit SymbolHelper(std::vector<std::string> symbols_file_directories)
-      : symbols_file_directories_(std::move(symbols_file_directories)){};
+  explicit SymbolHelper(std::vector<path> symbols_file_directories, path cache_directory)
+      : symbols_file_directories_(std::move(symbols_file_directories)),
+        cache_directory_(std::move(cache_directory)){};
 
-  [[nodiscard]] ErrorMessageOr<orbit_grpc_protos::ModuleSymbols> LoadUsingSymbolsPathFile(
-      const std::string& module_path, const std::string& build_id) const;
-  [[nodiscard]] ErrorMessageOr<orbit_grpc_protos::ModuleSymbols> LoadSymbolsFromFile(
-      const std::string& file_path, const std::string& build_id) const;
+  [[nodiscard]] ErrorMessageOr<path> FindSymbolsWithSymbolsPathFile(
+      const path& module_path, const std::string& build_id) const;
+  [[nodiscard]] ErrorMessageOr<path> FindSymbolsInCache(const path& module_path,
+                                                        const std::string& build_id) const;
+  [[nodiscard]] static ErrorMessageOr<orbit_grpc_protos::ModuleSymbols> LoadSymbolsFromFile(
+      const path& file_path);
 
-  [[nodiscard]] std::string GenerateCachedFileName(const std::string& file_path) const;
+  [[nodiscard]] path GenerateCachedFileName(const path& file_path) const;
 
  private:
-  const std::vector<std::string> symbols_file_directories_;
+  const std::vector<path> symbols_file_directories_;
```

Comment:
fs::path (it just looks strange having std:: stuff without ns prefix :)

Reply:
done -> see above